In [3]:
import pandas as pd 
data=pd.read_csv("CSA-Data.csv")
data.head

<bound method NDFrame.head of      "Children are safe among family members such as grandparents, uncles, aunts, cousins"  \
0                                                 Agree                                      
1                                                 Agree                                      
2                                              Disagree                                      
3                                                 Agree                                      
4                                                 Agree                                      
...                                                 ...                                      
2997                                              Agree                                      
2998                                              Agree                                      
2999                                              Agree                                      
3000                          

In [5]:
data.shape

(3002, 9)

There are 3002 rows and 9 columns(attributes)

In [8]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3002 entries, 0 to 3001
Data columns (total 9 columns):
 #   Column                                                                                                  Non-Null Count  Dtype 
---  ------                                                                                                  --------------  ----- 
 0   "Children are safe among family members such as grandparents, uncles, aunts, cousins"                   3002 non-null   object
 1   "Children are mainly abused by strangers in our society"                                                3002 non-null   object
 2   Male children dont need sexual abuse prevention knowledge                                               3002 non-null   object
 3   "Teaching sexual abuse prevention in school is not necessary. It will make children curious about sex"  3002 non-null   object
 4   Do you know what child grooming is?                                                              

In [10]:
data.isnull().sum()


"Children are safe among family members such as grandparents, uncles, aunts, cousins"                     0
"Children are mainly abused by strangers in our society"                                                  0
Male children dont need sexual abuse prevention knowledge                                                 0
"Teaching sexual abuse prevention in school is not necessary. It will make children curious about sex"    0
Do you know what child grooming is?                                                                       0
Do you know what signs to look for to identify if your child has been abused?                             0
Do you think children need post abuse counseling for recovering?                                          0
Do you think you should take legal action against the abuser of your child?                               0
Knowledge Level                                                                                           0
dtype: int64

Based on the above data, we can see that there are no missing values.

In [40]:
print(data.duplicated().sum())

2755


There are 2755 duplicated rows, and this might affect our analysis. We can remove them to ensure data quality.

In [55]:
data= data.drop_duplicates()

The above query is used to remove duplicate entries.

In [61]:
print(data.duplicated().sum())

0


The value (0) indicates that we have dropped duplicate entries.

In [48]:
data.describe()

"Children are safe among family members such as grandparents, uncles, aunts, cousins"  \
count                                                 247                                      
unique                                                  2                                      
top                                              Disagree                                      
freq                                                  124                                      

       "Children are mainly abused by strangers in our society"  \
count                                                 247         
unique                                                  2         
top                                              Disagree         
freq                                                  130         

       Male children dont need sexual abuse prevention knowledge  \
count                                                 247          
unique                                                  2          
top                                              Disagree          
freq                                                  137          

       "Teaching sexual abuse prevention in school is not necessary. It will make children curious about sex"  \
count                                                 247                                                       
unique                                                  2                                                       
top                                              Disagree                                                       
freq                                                  125                                                       

       Do you know what child grooming is?  \
count                                  247   
unique                                   2   
top                                     No   
freq                                   127   

       Do you know what signs to look for to identify if your child has been abused?  \
count                                                 247                              
unique                                                  2                              
top                                                    No                              
freq                                                  127                              

       Do you think children need post abuse counseling for recovering?  \
count                                                 247                 
unique                                                  2                 
top                                                   Yes                 
freq                                                  146                 

       Do you think you should take legal action against the abuser of your child?  \
count                                                 247                            
unique                                                  2                            
top                                                   Yes                            
freq                                                  137                            

       Knowledge Level  
count              247  
unique               2  
top           Beginner  
freq               181

There is an overwhelming number of people who think that children need post abuse counselling for recovering and this is good for sociaty. 

In [64]:
data["Do you think you should take legal action against the abuser of your child?"].value_counts()

Do you think you should take legal action against the abuser of your child?
Yes    137
No     110
Name: count, dtype: int64

**Yes:** 137 responses

**No:**  110 responses

Total Responses: 247

**Key Observations:**
Majority Support Legal Action:

137 responses **(55.5%)** indicate that a slight majority believes legal action should be taken against the abuser.

A Significant Minority **(44.5%)** Opposes Legal Action:

110 responses suggest that a substantial portion of people may not support taking legal action, possibly due to:

Fear of retaliation or escalation.

Cultural beliefs favoring privacy or family resolution over legal involvement.

Lack of trust in the legal system or belief that it may not lead to justice.

In [50]:
data['Male children dont need sexual abuse prevention knowledge'].value_counts()

Male children dont need sexual abuse prevention knowledge
Disagree    137
Agree       110
Name: count, dtype: int64

**Disagree:** 137 responses

**Agree:** 110 responses

Total Responses: 247

**Key Observations:**
**Majority Disagree:**

137 responses **(55.5%)** disagree with the statement, meaning they believe male children do need sexual abuse prevention knowledge. This indicates awareness that boys are equally vulnerable to sexual abuse and need proper education.

**A Substantial Minority Agree:**

110 responses **(44.5%)** agree, suggesting some still hold the view that male children do not need specific prevention education, possibly due to traditional beliefs or misconceptions about male vulnerability to sexual abuse.

**Implications:**
**Awareness Gap:** The responses suggest that while the majority recognizes the need for sexual abuse prevention for boys, a significant portion still holds outdated or incorrect beliefs about the risks boys face.

**Need for Education:** This highlights the importance of promoting inclusive sexual abuse prevention programs for both genders, as male children are also at risk.


In [52]:
data.groupby('Knowledge Level')['"Children are safe among family members such as grandparents, uncles, aunts, cousins"'].describe()

count unique       top freq
Knowledge Level                            
Beginner          181      2     Agree   96
Intermediate       66      2  Disagree   39

Beginners (1,711 respondents): Most common response is "Agree" (969 responses, ~56.6%).

Intermediates (1,291 respondents): Most common response is "Disagree" (890 responses, ~68.9%).


Beginners tend to agree more, possibly due to limited knowledge or exposure.

Intermediates disagree more, suggesting that increased knowledge leads to a shift in perspective.

There may be a correlation between knowledge level and critical thinking on the topic.

In [70]:
cross_tab = pd.crosstab(data['Male children dont need sexual abuse prevention knowledge'], data['Knowledge Level'])
print(cross_tab)

Knowledge Level                                     Beginner  Intermediate
Male children dont need sexual abuse prevention...                        
Agree                                                     88            22
Disagree                                                  93            44


**Beginners (181 total):**

**Agree:** 88 (48.2% of Beginner respondents)

**Disagree:** 93 (51.8% of Beginner respondents)

**Intermediates (66 total):**

**Agree:** 22 (33.3% of Intermediate respondents)

**Disagree:** 44 (66.7% of Intermediate respondents)

**Key Insights:**
**Beginners:** Slightly more than half disagree with the statement (51.8%), but a significant portion still agrees (48.2%).

**intermediates:** A stronger disagreement (66.7%) is observed among those with more knowledge, showing that increased knowledge results in a greater understanding of the need for abuse prevention education for boys.

**Conclusion:**
Knowledge impacts perspective: As respondents' knowledge level increases, they are more likely to reject the idea that male children don't need abuse prevention education.

In [37]:
print(data['Knowledge Level'].value_counts(normalize=True))  

Knowledge Level
Beginner        0.569953
Intermediate    0.430047
Name: proportion, dtype: float64


56% of the people have beginner knowledge level.

**CONCLUSION**


**Awareness of Family Abuse:** The disagreement with the idea that children are safe among family members indicates that respondents are aware of the risks that exist even within families.

**Strangers as Abusers:** The disagreement with the idea that children are mainly abused by strangers reflects a broader understanding that the abusers are often close to the child.

**Male Children and Prevention Knowledge:** The disagreement with the notion that male children don't need sexual abuse prevention shows growing recognition that both genders require education on this issue.

**Sexual Abuse Prevention in Schools:** The disagreement with the statement that teaching sexual abuse prevention in schools will cause curiosity about sex suggests that respondents support preventative education as crucial to children's safety.

**Knowledge Gaps in Grooming and Abuse Signs:** The "No" responses to knowing about child grooming and identifying signs of abuse indicate a need for further education and awareness in these areas.

**Post-Abuse Counseling and Legal Action:** The agreement that children need post-abuse counseling and that legal action should be taken against abusers reflects strong advocacy for victim support and accountability.

This data reveals that while there is significant awareness of the need for prevention, support, and legal action, there are also notable gaps in knowledge regarding specific aspects like child grooming and identifying signs of abuse. The results suggest the importance of expanding education on these topics, especially in communities where such knowledge is lacking.

